ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVMNASNetEncoder"]["model_variant"],
     )
 
-    channel_dim: int = schema_utils.PositiveInteger(
-        default=256,
-        description="Number of channels in hidden layer.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["channel_dim"],
+
+@register_encoder_config("mobilenetv2_torch", IMAGE)
+@dataclass
+class TVMobileNetV2EncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["mobilenetv2_torch"],
+        default="mobilenetv2_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    num_layers: int = schema_utils.PositiveInteger(
-        default=8,
-        description="The depth of the network (the number of Mixer blocks).",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["num_layers"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["base"],
+        default="base",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVMobileNetV2Encoder"]["model_variant"],
     )
 
-    avg_pool: bool = schema_utils.Boolean(
-        default=True,
-        description="If true, pools output over patch dimension, outputs a vector of shape (embed_size). If false, "
-        "the output tensor is of shape (n_patches, embed_size), where n_patches is img_height x img_width "
-        "/ patch_size².",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["avg_pool"],
+
+@register_encoder_config("mobilenetv3_torch", IMAGE)
+@dataclass
+class TVMobileNetV3EncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["mobilenetv3_torch"],
+        default="mobilenetv3_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        [
+            "small",
+            "large",
+        ],
+        default="small",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVMobileNetV3Encoder"]["model_variant"],
+    )
 
-@register_encoder_config("vit", IMAGE)
-@dataclass(repr=False)
-class ViTEncoderConfig(BaseEncoderConfig):
 
+@register_encoder_config("regnet_torch", IMAGE)
+@dataclass
+class TVRegNetEncoderConfig(TVBaseEncoderConfig):
     type: str = schema_utils.StringOptions(
-        ["vit"],
-        default="vit",
+        ["regnet_torch"],
+        default="regnet_torch",
         allow_none=False,
         description="Type of encoder.",
     )
 
-    height: int = schema_utils.NonNegativeInteger(
-        default=None,
-        description="Height of the input image.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["height"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
removed optional